packages/arb-bridge-eth/contracts/rollup/Rollup.sol

```diff
@@ -0,0 +1,699 @@
+            removeZombie(zombieNum);
+        } else {
+            zombieUpdateLatestStakedNode(zombieNum, latestStakedNode);
+        }
+    }
+
+    /**
+     * @notice Remove any zombies whose latest stake is earlier than the first unresolved node
+     * @param startIndex Index in the zombie list to start removing zombies from (to limit the cost of this transaction)
+     */
+    function removeOldZombies(uint256 startIndex) public {
+        uint256 currentZombieCount = zombieCount();
+        uint256 firstUnresolved = firstUnresolvedNode();
+        for (uint256 i = startIndex; i < currentZombieCount; i++) {
+            while (zombieLatestStakedNode(i) < firstUnresolved) {
+                removeZombie(i);
+                currentZombieCount--;
+                if (i >= currentZombieCount) {
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice Calculate the current amount of funds required to place a new stake in the rollup
+     * @return The current minimum stake requirement
+     */
+    function currentRequiredStake() public view returns (uint256) {
+        uint256 MAX_INT = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+        uint256 latestConfirmedDeadline = getNode(latestConfirmed()).deadlineBlock();
+        if (block.number < latestConfirmedDeadline) {
+            return baseStake;
+        }
+        uint256 latestConfirmedAge = block.number - latestConfirmedDeadline;
+        uint256 challengePeriodsPassed = latestConfirmedAge / challengePeriodBlocks;
+        if (challengePeriodsPassed > 255) {
+            challengePeriodsPassed = 255;
+        }
+        uint256 multiplier = 2**challengePeriodsPassed - 1;
+        if (multiplier == 0) {
+            multiplier = 1;
+        }
+
+        if (multiplier > MAX_INT / baseStake) {
+            return MAX_INT;
+        }
+
+        return baseStake * multiplier;
+    }
+
+    /**
+     * @notice Calculate the minimum time between assertions
+     * @return The minimum time between assertions in blocks
+     */
+    function minimumAssertionPeriod() public view returns (uint256) {
+        return challengePeriodBlocks / 10;
+    }
+
+    /**
+     * @notice Calculate the number of zombies staked on the given node
+     * @param node The node on which to count staked zombies
+     * @return The number of zombies staked on the node
+     */
+    function countStakedZombies(INode node) public view returns (uint256) {
+        uint256 currentZombieCount = zombieCount();
+        uint256 stakedZombieCount = 0;
+        for (uint256 i = 0; i < currentZombieCount; i++) {
+            if (node.stakers(zombieAddress(i))) {
+                stakedZombieCount++;
+            }
+        }
+        return stakedZombieCount;
+    }
+
+    /**
+     * @notice Verify that no stake has been placed within the last challenge period
+     */
+    function checkNoRecentStake() public view {
```

Comment:
Rather than naming functions like this `checkFoo`, it seems better to name them `requireFoo`.

Reply:
Good call. I've updated a bunch of those